src/utils.ts

```diff
@@ -464,17 +466,26 @@ function streamStats(
 `;
 }
 
+/**
+ * Used as a clean way to create a new monitor if it doesn't exist, otherwise uses the existing one.
+ */
 async function withMonitor<T>(
   mon: Monitor<RWLockWriter> | undefined,
   lockBox: LockBox<RWLockWriter>,
   lockConstructor: { new (): RWLockWriter },
   fun: (mon: Monitor<RWLockWriter>) => Promise<T>,
   locksPending?: Map<string, { count: number }>,
 ): Promise<T> {
-  const _mon = mon ?? new Monitor(lockBox, lockConstructor, locksPending);
-  const result = await fun(_mon);
-  if (mon != null) await _mon.unlockAll();
-  return result;
+  if (mon == null) {
```

Comment:
The `fun` should be `f`, be consistent.

Reply:
Fixing, Also this block of code could've been a single review comment.

Reply:
Fixed.